includes/fbutils.php

```diff
@@ -90,12 +90,25 @@ public static function make_url( $url ) {
 		 * @return string
 		 */
 		public static function get_fb_retailer_id( $woo_product ) {
-			$woo_id = $woo_product->get_id();
-
 			// Call $woo_product->get_id() instead of ->id to account for Variable
 			// products, which have their own variant_ids.
-			return $woo_product->get_sku() ? $woo_product->get_sku() . '_' .
-			$woo_id : self::FB_RETAILER_ID_PREFIX . $woo_id;
+			$woo_id = $woo_product->get_id();
+			$sku = $woo_product->get_sku();
+			$retailer_id = $sku ? $sku . '_' . $woo_id : self::FB_RETAILER_ID_PREFIX . $woo_id;
+
+			switch ( get_option( \WC_Facebookcommerce_Integration::SETTING_FB_RETAILER_ID_TYPE )) {
+				case \WC_Facebookcommerce_Integration::FB_RETAILER_ID_TYPE_SKU:
+					$retailer_id = $sku;
```

Comment:
If the customer selected `FB_RETAILER_ID_TYPE_SKU` but a product doesn't have a SKU the `$retailer_id` variable ends up being empty. I think we should add a fallback here.

Maybe good idea to add a couple of tests to ensure we get a proper value in those cases as well.

Reply:
Yeah, I had thought of this. However, rather than trying to define the behavior here (pick an alt. field, for example, which could potentially cause more confusion) I thought it would actually be better to just let Facebook reject the product at the catalog sync level.

Reply:
Got it. I think it makes sense the way you put it, but may not work that way right now because those rejections are very hard to detect for a user. The Batch API doesn't fail right away, instead you have to use the returned handle to check the status of the request and see a possible list of errors, and the plugin doesn't do that.

Probably good idea to add a FAQ entry or something similar in the documentation to instruct both merchants and support teams to check the new settings and the configuration of affected products when someone reports that some products are not syncing.